### DISCUSSION TITLE
Improve programmatic accessibility

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/klog#376] [Feature ideas & discussions]
I’m currently experimenting with wrapping shell scripts around klog, e.g. to automatically create new `.klg` files based on a date pattern. Say, you follow the convention `YYYY-MM.klg` then such a shell script could take care of creating a new file on the fly at the beginning of a new month. (Just for reference: incorporating such functionality directly into klog is out of scope.)

In order to improve that kind of programatic access to klog, the following things would be handy:

- Distinct error codes, so that certain error cases can be reliably distinguished (e.g. “file not found“). Currently, all errors cause klog to exit with `-1`.
- A “quiet” output mode, to only print the raw “response” of a command without any labels meant to structure the text for humans, e.g.:
  `klog bookmark get` → `Current bookmark: /home/users/foo/myfile.klg`
  `klog bookmark get --quiet` → `/home/users/foo/myfile.klg`
  That way you can do `CURRENT_BOOKMARK=$(klog bookmark get --quiet)` in a more robust way.
  - This flag only makes sense on certain subcommands, e.g. `total`, `bookmark` or `version`
  - `--quiet` should imply `--no-color`

**Comment by vladdeSV** [comment]
Could `klog version --quiet` return an easily parseable value, such as `v2.4`?

**Comment by jotaen** [comment]
@vladdeSV yes, have implemented it that way now
